os/kernel/semaphore/sem_tickwait.c

```diff
@@ -180,7 +180,7 @@ int sem_tickwait(FAR sem_t *sem, systime_t start, uint32_t delay)
 
 	ret = sem_wait(sem);
 	if (ret < 0) {
-		/* Return the errno from sem_wait() */
+		/* Return the errno from sem_wait() or from sem_timeout() */
```

Comment:
If you see line no 185, we are using get_errno() API to get the error no and then assign it to return variable "ret" ,   so it means  anybody can set the errno using set_errno()  .  In this particular case   error no can be set via sem_timeout also as  "TIMEOUT"   and sem_wait just return  OK or ERROR .   As you said, sem_tickwait gets return value only from sem_wait as OK or ERROR  , however  setting the correct error code is done either by  sem_wait or sem_timeout.  So I modify the comment saying "return the errorno from sem_wait or sem_timeout"